subsys/bluetooth/audio/vcs.c

```diff
@@ -671,243 +721,328 @@ int bt_vcs_mute(struct bt_conn *conn)
+		return -EINVAL;
+	}
+
 	if (IS_ENABLED(CONFIG_BT_VCS_CLIENT_VOCS) &&
-	    conn && bt_vcs_client_valid_vocs_inst(conn, inst)) {
+	    bt_vcs_client_valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_location_set(inst, location);
 	}
 
-	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && !conn && valid_vocs_inst(inst)) {
+	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_location_set(inst, location);
 	}
 
 	return -EOPNOTSUPP;
 }
 
-int bt_vcs_vocs_state_set(struct bt_conn *conn, struct bt_vocs *inst,
+int bt_vcs_vocs_state_set(struct bt_vcs *vcs, struct bt_vocs *inst,
 			  int16_t offset)
 {
+	CHECKIF(vcs == NULL) {
+		BT_DBG("NULL vcs instance");
+		return -EINVAL;
+	}
+
 	if (IS_ENABLED(CONFIG_BT_VCS_CLIENT_VOCS) &&
-	    conn && bt_vcs_client_valid_vocs_inst(conn, inst)) {
+	    bt_vcs_client_valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_state_set(inst, offset);
 	}
 
-	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && !conn && valid_vocs_inst(inst)) {
+	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_state_set(inst, offset);
 	}
 
 	return -EOPNOTSUPP;
 }
 
-int bt_vcs_vocs_description_get(struct bt_conn *conn, struct bt_vocs *inst)
+int bt_vcs_vocs_description_get(struct bt_vcs *vcs, struct bt_vocs *inst)
 {
+	CHECKIF(vcs == NULL) {
+		BT_DBG("NULL vcs instance");
+		return -EINVAL;
+	}
+
 	if (IS_ENABLED(CONFIG_BT_VCS_CLIENT_VOCS) &&
-	    conn && bt_vcs_client_valid_vocs_inst(conn, inst)) {
+	    bt_vcs_client_valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_description_get(inst);
 	}
 
-	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && !conn && valid_vocs_inst(inst)) {
+	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_description_get(inst);
 	}
 
 	return -EOPNOTSUPP;
 }
 
-int bt_vcs_vocs_description_set(struct bt_conn *conn, struct bt_vocs *inst,
+int bt_vcs_vocs_description_set(struct bt_vcs *vcs, struct bt_vocs *inst,
 				const char *description)
 {
+	CHECKIF(vcs == NULL) {
+		BT_DBG("NULL vcs instance");
+		return -EINVAL;
+	}
+
 	if (IS_ENABLED(CONFIG_BT_VCS_CLIENT_VOCS) &&
-	    conn && bt_vcs_client_valid_vocs_inst(conn, inst)) {
+	    bt_vcs_client_valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_description_set(inst, description);
 	}
 
-	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && !conn && valid_vocs_inst(inst)) {
+	if (IS_ENABLED(CONFIG_BT_VCS_VOCS) && valid_vocs_inst(vcs, inst)) {
 		return bt_vocs_description_set(inst, description);
 	}
 
 	return -EOPNOTSUPP;
 }
 
-int bt_vcs_aics_state_get(struct bt_conn *conn, struct bt_aics *inst)
+int bt_vcs_aics_state_get(struct bt_vcs *vcs, struct bt_aics *inst)
 {
+	CHECKIF(vcs == NULL) {
+		BT_DBG("NULL vcs instance");
+		return -EINVAL;
+	}
+
 	if (IS_ENABLED(CONFIG_BT_VCS_CLIENT_AICS) &&
-	    conn && bt_vcs_client_valid_aics_inst(conn, inst)) {
-		return bt_aics_state_get(conn, inst);
+	    bt_vcs_client_valid_aics_inst(vcs, inst)) {
+		return bt_aics_state_get(vcs->cli.conn, inst);
```

Comment:
Does AICS still use the connection pointer here?  (I have not reviewed the AICS PR yet.)

Reply:
Yes, but the AICS PR will change that, so whichever one of those 2 PRs gets merged, the other will have some conflicts.